crates/ra_syntax/src/ast.rs

```diff
@@ -115,19 +115,30 @@ pub trait DocCommentsOwner: AstNode {
     }
 
     /// Returns the textual content of a doc comment block as a single string.
-    /// That is, strips leading `///` and joins lines
+    /// That is, strips leading `///` (+ optional 1 character of whitespace)
+    /// and joins lines.
     fn doc_comment_text(&self) -> std::string::String {
         self.doc_comments()
             .filter(|comment| comment.is_doc_comment())
             .map(|comment| {
-                let prefix = comment.prefix();
-                let trimmed = comment
-                    .text()
-                    .as_str()
-                    .trim()
-                    .trim_start_matches(prefix)
-                    .trim_start();
-                trimmed.to_owned()
+                let prefix_len = comment.prefix().len();
+
+                // Strip leading and trailing whitespace
+                let line = comment.text().as_str().trim();
```

Comment:
I think we should not do arbitrary whitespace trimming here and rather preserve whitespace as the user typed it.

The logic should be "if the string starts with preffix + single space, trim that, otherwise trim preffix"